src/analysis/stack-lattice.h

```diff
@@ -0,0 +1,139 @@
+#ifndef wasm_analysis_stack_lattice_h
+#define wasm_analysis_stack_lattice_h
+
+#include <deque>
+
+#include "lattice.h"
+
+namespace wasm::analysis {
+
+template<typename StackFrameLattice> class StackLattice {
+  static_assert(is_lattice<StackFrameLattice>);
+  StackFrameLattice& stackFrameLattice;
+
+public:
+  StackLattice(StackFrameLattice& stackFrameLattice)
+    : stackFrameLattice(stackFrameLattice) {}
+
+  class Element {
+    std::deque<typename StackFrameLattice::Element> stackValue;
+    bool top = false;
```

Comment:
How about using a `std::optional<std::deque<...>>` where top is represented by `std::nullopt`? That way we don't have to worry about the deque hanging out in top values.